homeassistant/components/elmo_alarm/__init__.py

```diff
@@ -0,0 +1,219 @@
+"""Support for e-connect Elmo alarm system."""
+from collections import namedtuple
+from datetime import timedelta
+import logging
+
+from elmo.api.client import ElmoClient
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_SCAN_INTERVAL,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_call_later
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "elmo_alarm"
+CLIENT = "client"
+USERNAME = "username"
+PASSWORD = "password"
+STATES = "states"
+STATE_NAME = "name"
+STATE_ZONES = "zones"
+
+CONF_VENDOR = "vendor"
+CONF_USER = "username"
+CONF_PASS = "password"
+CONF_STATES = "states"
+CONF_STATE_NAME = "name"
+CONF_ZONES = "zones"
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+
+SIGNAL_ZONE_CHANGED = "elmo_alarm.zone_changed"
+SIGNAL_INPUT_CHANGED = "elmo_alarm.input_changed"
+SIGNAL_ARMING_STATE_CHANGED = "elmo_alarm.arming_state_changed"
+
+INPUT_TYPE = "opening"
+ZONE_TYPE = "safety"
+
+ZONE_ARMED = 0
+ZONE_DISARMED = 1
+
+INPUT_ALERT = 1
+INPUT_WAIT = 0
+
+ZoneData = namedtuple("ZoneData", ["zone_id", "zone_name", "state"])
+InputData = namedtuple("InputData", ["input_id", "input_name", "state"])
+
+# DEFAULT_ZONE_TYPE = "motion"
+STATE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_STATE_NAME): cv.string,
+        vol.Required(CONF_ZONES): vol.All(cv.ensure_list),
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Required(CONF_VENDOR): cv.string,
+                vol.Required(CONF_USER): cv.string,
+                vol.Required(CONF_PASS): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): vol.All(cv.time_period, cv.positive_timedelta),
+                vol.Required(CONF_STATES): vol.All(cv.ensure_list, [STATE_SCHEMA]),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+class ElmoClientWrapper(ElmoClient):
+    def __init__(self, host, vendor, username, password, states_config):
+        self._username = username
+        self._password = password
+        self._states_config = states_config
+        self._data = None
+        self._states = None
+        self._state = None
+
+        ElmoClient.__init__(self, host, vendor)
+
+    async def update(self):
+        try:
+            data = self.check()
+        except Exception:
```

Comment:
Please catch the specific exception that could be thrown.

Reply:
Exception PermissionDenied is catched and logged.